### ISSUE TITLE
Mod is incompatible with other mods that add timestamps

### ISSUE THREAD

**Issue by 70CentsApple** [caoimhebyrne/compact-chat#23]
Some mods (e.g. Tweakeroo, Chat Tools etc.) adds a timestamp in front of the messages, however it doesn't work very well with Compact Chat. Also, I found a former issue that is similar to my problem that is https://github.com/caoimhebyrne/compact-chat/issues/21.

I can give you a RegEx that can matches Timestamps like [12:34:56] or [12:34], that is `\[?\d{1,2}:\d{2}(:\d{2})*\]? *` .

**Comment by caoimhebyrne** [comment]
I'll take a look at this again. Thanks for the report!

**Comment by caoimhebyrne** [comment]
**TL;DR: I have to look into this a bit more to see if I can find a nice solution. Sorry about the delay on this!**

Tweakeroo, and other mods like that, do a bit of manipulation on the text (a lot more than I'd like them to), which makes it quite hard to compare the text when removing the timestamps.

The main issue with that is, our mixin runs before Tweakeroo's (probably due to the name of our mixin `ChatHudMixin` vs theirs `MixinChatHud`, so the message that we first get (to append the occurrences count) doesn't contain the timestamp. The only time we see the message with the timestamp, is when we go to remove old occurrences of the message. This is *fine*, but again, not ideal, as they modify the hierarchy of the message a bit, making it hard to see if the messages are equal after removing the timestamp.

A `Text` component in Minecraft comes in many forms, and can have many different properties. 

### A message, modified by Tweakeroo, with the timestamp stripped by Compact Chat
![image](https://github.com/caoimhebyrne/compact-chat/assets/71222289/3be5b4d7-3c3b-4fc2-8b54-b39265f093d6)

### A normal chat message
![image](https://github.com/caoimhebyrne/compact-chat/assets/71222289/570891c2-ec92-4295-81ef-b24603ed7347)

> Yes, I've tried just comparing them with a `getString`, that brings suboptimal results.

**Comment by 70CentsApple** [comment]
> 

no matter, you can finish it whenever you like :)

**Comment by caoimhebyrne** [comment]
I've made an upstream patch to Tweakeroo to help add support for CompactChat, it will also need changes on my side. If there's any other mods that you're aware of, can you make sure they follow similar functionality to what I've demonstrated in my patch?

https://github.com/maruohon/tweakeroo/pull/447

**Comment by 70CentsApple** [comment]
> I've made an upstream patch to Tweakeroo to help add support for CompactChat, it will also need changes on my side. If there's any other mods that you're aware of, can you make sure they follow similar functionality to what I've demonstrated in my patch?
> 
> [maruohon/tweakeroo#447](https://github.com/maruohon/tweakeroo/pull/447)

yeah thank you

**Comment by caoimhebyrne** [comment]
I think ChatTools should be compatible with this now, I've committed it to another branch until I do some more testing, see 3568670.

https://github.com/70CentsApple/ChatTools/blob/1.20/src/main/java/net/apple70cents/chattools/mixin/ChatHudMixin.java

**Comment by caoimhebyrne** [comment]
Going to keep this open until the release is public.

**Comment by caoimhebyrne** [comment]
This has now been published to [Modrinth](https://modrinth.com/mod/compact-chat/version/2.0.2) and [GitHub Releases](https://github.com/caoimhebyrne/compact-chat/releases/tag/2.0.2).

**Comment by 70CentsApple** [comment]
Hey! I'm sorry to inform you that such issue can be easily reproduced once again in MC 1.20.4 due to some probable reasons. However, I'm glad to inform that I have already figure it out and I could make a PR for that! @caoimhebyrne
